File: Coding_Training_Camp/PTT_2024-03-16/P2/P2.py
while 1:
    n=int(input())
    if n==0:
        break
    print(f"Original number was {n}")
    chain=[]
    while True:
        if n in chain:
            break
        chain.append(n)
        small = sorted(str(n))
        big = small[::-1]
        first = int(''.join(big))
        second = int(''.join(small))
        print(f"{first} - {second} = {first - second}")
        n=first-second
    print(f"Chain length {len(chain)}\n")
#註解
#注意:循環的部分有可能是超過一個以上的數字
#所以要用一個list來存放每次的數字 
#而不是直接使用前一個數字來做比較    

# def get_next_number(n):
#     small = sorted(str(n))
#     big = small[::-1]
#     first = int(''.join(big))
#     second = int(''.join(small))
#     print(f"{first} - {second} = {first - second}")
#     return first - second

# def solve_uva_263(n):
#     chain = []
#     while True:
#         if n in chain:
#             break
#         chain.append(n)
#         n = get_next_number(n)
#     return chain

# while True:
#     n = int(input())
#     if n == 0:
#         break
#     print(f"Original number was {n}")
#     chain = solve_uva_263(n)
#     print(f"Chain length {len(chain)}\n")
    

File: Coding_Training_Camp/PTT_2024-03-16/P2/P2.cpp
#include <iostream>
#include <string>
#include <algorithm>
#include <vector>

using namespace std;

int main() {
    while (1) {
        int n;
        cin >> n;
        if (n == 0) {
            break;
        }
        int now = n;
        int last = 0;
        cout << "Original number was " << n << endl;
        int count = 0;
        while (last != now) {
            int temp = now;
            last = now;
            vector<int> small;
            if (temp == 0) {
                small.push_back(0);
            }
            while (temp != 0) {
                small.push_back(temp % 10);
                temp /= 10;
            }
            sort(small.begin(), small.end());
            vector<int> big = small;
            reverse(big.begin(), big.end());
            string first_str = "";
            string second_str = "";
            for (int i : big) {
                first_str += to_string(i);
            }
            for (int i : small) {
                second_str += to_string(i);
            }
            int first = stoi(first_str);
            int second = stoi(second_str);
            now = first - second;
            cout << first << " - " << second << " = " << now << endl;
            count++;
        }
        cout << "Chain length " << count << "\n" << endl;
    }
    return 0;
}